### PR TITLE
feat(direnv): experimental SENTRY_PYTHON3

### PR DESCRIPTION
**Pull request by evanpurkhiser**
`direnv` now supports a python3 environment by setting the `SENTRY_PYTHON3` flag.

This uses the `.venv3` directory for the virtual env. This makes the assumption that we'll have our tooling support both `.venv` and `.venv3`.

If we don't want to do this, I can definitely have it do some fancy symlink things. But I suspect symlinks are going to cause us more trouble than we might expect.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -4,7 +4,10 @@
 import os
 import sys
 
-if not os.environ.get("SENTRY_PYTHON3") and sys.version_info[:2] != (2, 7):
+if os.environ.get("SENTRY_PYTHON3") and sys.version_info[:2] != (3, 6):
+    sys.exit("Error: Sentry [In EXPERIMENTAL python 3 mod] requires Python 3.6.")
+
+elif sys.version_info[:2] != (2, 7):
```
Comment (EvanPurkhiser) [inline setup.py:10]: Oops, that's wrong

File: .envrc [hunk 2]
```diff
@@ -85,68 +95,79 @@ export PIP_DISABLE_PIP_VERSION_CHECK=on
...
     info "Failed to activate virtualenv. Your virtualenv's probably symlinked." \
-    "We want everyone to be on the same page here, so you'll have to recreate your virtualenv."
-    advice_init_venv
+        "We want everyone to be on the same page here, so you'll have to recreate your virtualenv."
+    advice_init_venv "$python_version" "$venv_name"
 fi
 
-python -c "import sys; sys.exit(sys.version_info[:2] != (2, 7))" || \
-    die "For some reason, the virtualenv isn't Python 2.7."
+if [ "x$SENTRY_PYTHON3"  = "x1" ]; then
```
Comment (joshuarli) [inline .envrc:156]: remove the x idk why people do that lol

File: pyproject.toml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools>=40.2.0", "wheel"]
+requires = ["setuptools>=40.2.0,<46.0.0", "wheel"]
```
Comment (joshuarli) [inline pyproject.toml:2]: Why?

Comment (joshuarli) [reply pyproject.toml:2]: Or rather, what's using this? https://github.com/pypa/setuptools/issues/2017#issuecomment-605354361

Comment (EvanPurkhiser) [reply pyproject.toml:2]: Maxmindb

Comment (joshuarli) [reply pyproject.toml:2]: Could you provide more detail? And preferably in comment form.

File: scripts/ensure-venv.sh [hunk 4]
```diff
@@ -6,22 +6,36 @@ if [ "x$SENTRY_NO_VIRTUALENV_CREATION" == "x1" ]; then
...
+
+if [ "x$SENTRY_PYTHON3" = "x1" ]; then
+    venv_name=".venv3"
+    python_bin="python3.6"
+fi
+
+die() {
+    cat >&2 "${@}"
+    exit 1
+}
```
[the remark this answers is not in the document]
Comment (joshuarli) [reply scripts/ensure-venv.sh:20]: I don't think that's necessary.

File: pyproject.toml (hunk 3 above)
Comment (EvanPurkhiser) [reply pyproject.toml:2]: Yeah I can

File: scripts/ensure-venv.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (EvanPurkhiser) [reply scripts/ensure-venv.sh:20]: I mean the python 3 related variables.

File: .envrc (hunk 2 above)
Comment (joshuarli) [reply .envrc:156]: Posterity: apparently really old shells aren't good with empty strings. This isn't a concern in 2020 though. Well. One can only hope.

File: pyproject.toml (hunk 3 above)
Comment (EvanPurkhiser) [reply pyproject.toml]: Ok, I can't seem to reproduce this now, so just going to remove the constraint and see if it flares up later.

File: .envrc [hunk 5]
```diff
@@ -85,68 +95,79 @@ export PIP_DISABLE_PIP_VERSION_CHECK=on
...
 
-
 ### Python ###
 
+# we're enforcing that virtualenv be in .venv{,3}, since future tooling e.g.
+# venv-update will rely on this.
+venv_name=".venv"
+python_version="python2.7"
+
+if [ "$SENTRY_PYTHON3" = "1" ]; then
```
Comment (joshuarli) [inline .envrc:125]: I'd just test that this env var is set to something. = 1 is unnecessarily strict. You missed changing the other places from x1, so might as well just hit 2 birds with 1 stone there.

**Review by joshuarli — APPROVED** [review]
lgtm, see nit https://github.com/getsentry/sentry/pull/19926/files#r458409235

File: .envrc (hunk 5 above)
Comment (EvanPurkhiser) [reply .envrc:125]: hmm, I would prefer some kind of truthy check, otherwise if you set it to `0` it will still work.

**Pull request merged**